cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/cask360/Cask360Entity.java

```diff
@@ -0,0 +1,361 @@
+import co.cask.cdap.api.dataset.lib.cask360.Cask360GroupData.Cask360GroupDataTime;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+import com.google.gson.JsonSyntaxException;
+
+import java.lang.reflect.Type;
+import java.text.ParseException;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * Represents a single entity in a {@link Cask360Table}.
+ * <p>
+ * An entity is represented by a unique String id and any number of groups of
+ * data about the entity. Each group of data is identified by a String name and
+ * a {@link Cask360Group} object.
+ */
+public class Cask360Entity implements Comparable<Cask360Entity> {
+
+  /** Gson with custom serialization */
+  private static final Gson gson = Cask360Entity.getGson();
+
+  /** Entity ID */
+  private String id;
+
+  /** Map from group name to group data */
+  Map<String, Cask360Group> groups;
+
+  /**
+   * Constructs an empty entity with the specified ID.
+   *
+   * @param id
+   *          unique identifier of the entity
+   */
+  public Cask360Entity(String id) {
+    this(id, new TreeMap<String, Cask360Group>());
+  }
+
+  /**
+   * Constructs a new instance of a {@link Cask360Entity}. References the
+   * specified map, does not create an internal copy.
+   *
+   * @param id
+   *          the id of the entity
+   * @param groups
+   *          groups of data of the entity
+   */
+  public Cask360Entity(String id, Map<String, Cask360Group> groups) {
+    this.id = id;
+    this.groups = groups;
+  }
+
+  /**
+   * Get the ID of the entity.
+   *
+   * @return unique identifier string of this entity
+   */
+  public String getID() {
+    return id;
+  }
+
+  /**
+   * Get the groups of data of the entity.
+   *
+   * @return map of group names to group data
+   */
+  public Map<String, Cask360Group> getGroups() {
+    return groups;
+  }
+
+  // Write Operations
+
+  /**
+   * Write the specified data key and value to the group with the specified
+   * name.
+   * <p>
+   * Assumes the group is of the type {@link Cask360GroupType#MAP}, otherwise
+   * will throw an exception.
+   *
+   * @param name
+   *          the group name
+   * @param key
+   *          the data key
+   * @param value
+   *          the data value
+   */
+  public void write(String name, String key, String value) {
+    Cask360Group group = this.groups.get(name);
+    if (group != null) {
+      group.getDataAsMap().put(key, value);
+    } else {
+      Cask360GroupDataMap dataMap = new Cask360GroupDataMap();
+      dataMap.put(key, value);
+      group = new Cask360Group(name, dataMap);
+      this.groups.put(name, group);
```

Comment:
We usually don't use `this.` to refer to class field (except in constructor). Although it's not a must change, but it aligns with the current codebase better.